index.js

```diff
@@ -521,6 +521,54 @@
       [$.Function, b, a, c],
       function(f, x, y) { return f(y, x); });
 
+  //# map :: Functor f => (a -> b) -> f a -> f b
+  //.
+  //. Takes a function and a [Functor][], applies the function to each of the
+  //. Functor's values, and returns a Functor of the same shape.
+  //.
+  //. ```javascript
+  //. > S.map(S.inc, [1, 2, 3])
+  //. [2, 3, 4]
+  //.
+  //. > S.map(S.inc, {a: 1, b: 2, c: 3})
+  //. {a: 2, b: 3, c: 4}
+  //.
+  //. > S.map(S.inc, S.Just(2))
+  //. S.Just(3)
+  //.
+  //. > S.map(S.inc, S.Nothing())
+  //. S.Nothing()
+  //.
+  //. > S.map(S.not, S.odd)(2)
+  //. true
+  //. ```
+  var map = S.map =
+  def('map',
+      {a: [Functor], b: [Functor]},
+      [$.Function, a, b],
+      function(f, functor) {
+        var type = _type(functor);
+        if (type === 'Array') {
+          var xs = [];
+          for (var idx = 0; idx < functor.length; idx += 1) {
+            xs.push(f(functor[idx]));
+          }
+          return xs;
+        } else if (type === 'Function') {
+          return compose(f, functor);
+        } else if (hasMethod('map')(functor)) {
+          return functor.map(f);
+        } else {
+          var obj = {};
+          for (var key in functor) {
+            if (Object.prototype.hasOwnProperty.call(functor, key)) {
```

Comment:
`S.keys` is next on my hitlist once https://github.com/plaid/sanctuary-def/pull/33 lands.